src/plugins/intel_cpu/src/emitters/snippets/cpu_runtime_configurator.cpp

```diff
@@ -18,20 +19,52 @@ void CPURuntimeConfigurator::update(const std::shared_ptr<ov::snippets::lowered:
     RuntimeConfigurator::update(linear_ir);
 
     if (linear_ir->is_dynamic()) {
+        const auto& loop_manager = linear_ir->get_loop_manager();
+        update_loop_args(loop_manager);
+        update_brgemms(loop_manager);
         get_kernel_executor_table()->update_state();
-        update_loop_args(linear_ir);
+    }
+}
+
+void CPURuntimeConfigurator::initialization(const std::shared_ptr<ov::snippets::lowered::LinearIR>& linear_ir) {
+    RuntimeConfigurator::initialization(linear_ir);
+
+    for (const auto& expr : *linear_ir) {
+        if (ov::is_type<ov::intel_cpu::BrgemmCPU>(expr->get_node())) {
+            const auto& in0_desc = expr->get_input_port_descriptor(0);
+            const auto& in1_desc = expr->get_input_port_descriptor(1);
+            const auto& out_desc = expr->get_output_port_descriptor(0);
+
+            const auto& in0_subtensor = in0_desc->get_subtensor();
+            const auto& in1_subtensor = in1_desc->get_subtensor();
+            const auto& out_subtensor = out_desc->get_subtensor();
+
+            // TODO [146125]: At the moment only blocking by dynamic M is supported
+            //                So we save Brgemm with only dynamic M
+            //                If there are other dynamic dimensions, throw exception for now
+            OPENVINO_ASSERT(!snippets::utils::is_dynamic_value(*in0_subtensor.crbegin()) &&
+                            !snippets::utils::is_dynamic_value(*in1_subtensor.crbegin()) &&
+                            !snippets::utils::is_dynamic_value(*(++in1_subtensor.crbegin())) &&
+                            !snippets::utils::is_dynamic_value(*out_subtensor.crbegin()),
+                            "CPURuntimeConfigurator supports only dynamic M in Brgemm subtensors");
+            OPENVINO_ASSERT(*(++in0_subtensor.crbegin()) == *(++out_subtensor.crbegin()),
+                            "Incorrect values in subtensors of BrgemmCPU");
```

Comment:
Do we really need to perform all this validation in runtime?
It looks like the Runtime configurator is not the right place to do it bot in terms of performance impact and responsibility domain.

Reply:
I left them here because of some reasons:
1. It's `initialization` stage of `RuntimeConfigurator` -> evaluate only once
2. These are temporary checks, because we add blocking by dynamic K, N soon
3. I decided to not create new validation pass from CPU Plugin backend to just validate these subtensors. Moreover, these checks will be removed soon.

Reply:
> Moreover, these checks will be removed soon

Then why wait? 😁 


Seriously, I think we'll have to remove them because we won't have this list of brgemm ops when subtensor setting will be delegated to  `BrgemmKernelExecutor::update_config`.
When I was writing this method, I thought about adding subtensor validation there, but skipped it because we had a dedicated pass for that. But now, when `update_config` would be modifying subtensors based on runtime parameters, it makes sense to add some sanity checks there.

Reply:
Removed as other Brgemm-related things from `CPURuntimeConfigurator`